File: renderer.cc
#include <SDL2_gfxPrimitives.h>
#include "renderer.hh"

#define RENDER_FLAGS SDL_RENDERER_ACCELERATED|\
                     SDL_RENDERER_PRESENTVSYNC

Renderer::Renderer(SDL_Window *win)
{
	ren = SDL_CreateRenderer(win, -1, RENDER_FLAGS);		
}

Renderer::~Renderer()
{
	SDL_DestroyRenderer(ren);
}

void
Renderer::render_circle(int x, int y, int radius, SDL_Color color)
{
	aacircleRGBA(ren, x, y, radius, color.r, color.g, color.b, 0xff);
	filledCircleRGBA(ren, x, y, radius, color.r, color.g, color.b, 0xff);
}

File: renderer.py
import pygame as pg

class Renderer:
	def __init__(self, surf):
		self.surf = surf;
	
	def render_circle(self, x, y, r, color):
		pg.draw.circle(self.surf, color, (x, y), r);

	def render_rectangle(self, x, y, w, h, color):
		pg.draw.rect(self.surf, color, (x, y, w,h));